src/doc/en/developer/github.rst

```diff
@@ -0,0 +1,438 @@
+
+
+.. _chapter-github:
+
+=============================
+The Sage Repository on GitHub
+=============================
+
+The center of Sage development is `the SageMath organization on GitHub
+<https://github.com/sagemath>`_, which consists of many repositories related
+with Sage. The most important one among them is of course `the Sage repository
+<https://github.com/sagemath/sage>`_, which we call "the Sage repo" for short.
+
+
+.. _section-github-account:
+
+Obtaining a GitHub account
+==========================
+
+To share your work on Sage, you need a GitHub account. If you do not have one
+yet, choose a username and `create an account <https://github.com/join>`_. In
+the following, we assume your username "alice". So you always read your own
+username if you see "alice".
+
+
+.. _chapter-github-cli:
+
+Using the GitHub CLI
+====================
+
+GitHub provides a command-line interface, the GitHub CLI, that can be used
+instead of the web interface.  The central component of the GitHub CLI is the
+``gh`` command that you can use in your terminal.
+
+Installation
+------------
+
+The page :ref:`spkg_github_cli` documents how to install the ``gh`` command for
+your platform. Or see `GitHub CLI <https://cli.github.com>`_ from GitHub.
+
+Configuration
+-------------
+
+You have to authenticate to your GitHub account to allow ``gh`` command to
+interact with GitHub. Typically the authorization proceeds as follows::
+
+    [alice@localhost sage]$ gh auth login
+    ? What is your preferred protocol for Git operations? HTTPS
+    ? Authenticate Git with your GitHub credentials? Yes
+    ? How would you like to authenticate GitHub CLI? Login with a web browser
+
+    ! First copy your one-time code: 3DA8-5ADA
+    Press Enter to open github.com in your browser...
+    ✓ Authentication complete.
+    - gh config set -h github.com git_protocol https
+    ✓ Configured git protocol
+    ✓ Logged in as sage
+
+where a web browser is used to enter credentials. You can also use an
+authentication token instead, in which case you must first generate `a Personal
+Access Token here <https://github.com/settings/tokens>`_.
+
+Next set the default repo for the ``gh`` command::
+
+    [alice@localhost sage]$ gh repo set-default sagemath/sage
+
+and check::
+
+    [alice@localhost sage]$ gh repo view
+    sagemath/sage
+    ...
+
+which will show the default repo along with its readme, which is quite long.
+
+
+Linking Git to your GitHub account
+==================================
+
+In order for your Git to work with GitHub, your GitHub account needs to be
+linked with your Git.  No action is needed if you have already contributed to
+any other project on GitHub and set up Git credentials or SSH keys for this.
+
+The above dialogue from ``gh auth login`` already linked your Git to GitHub by
+HTTPS protocol using your GitHub credentials. We assume this in the rest of
+this guide. For more details, see `Caching your GitHub credentials in Git
```

Comment:
I have no experience with GCM. It seems an advanced tool to manage user authentication information to remote repositories. Do we have to promote GCM to beginning Sage developers? For this guide, using "gh auth login" seems simplest and sufficient.

Reply:
GCM seems to be the most popular way to login in remote repos using git (besides for github it also works for say gitlab). It has also a bit more security features than gh auth, and comes with better WSL integration. But yeah, both are developed and supported by github.

Reply:
As another data point, on macOS I'm just using the builtin credential manager.

The "gh auth" method covers all platforms uniformly, which is why I would prefer it as the first thing to explain in our documentation.

Could you elaborate on what's missing on WSL?

Reply:
For WSL, GCM integrates with the credentials in the main windows system - this is not supported by gh.

What about moving this part into the "setup git" section, or just direct people to the github docs "Caching your GitHub credentials in Git...", then they can choose their own most convenient solution.

Reply:
"setup git" section is part of the "walkthrough", which is before "git interaction with github".

The guide already directs people to "Caching your GitHub credentials in Git..." for more information. They can choose.

But I think the guide should set forth an easy default way to authenticate git to github. "gh auth" is the easy default way.

As you say "GCM seems to be the most popular way", I will add a bit more about GCM instead of "for more details ...".